src/mixins/nft.js

```diff
@@ -256,39 +261,34 @@ export default {
     },
     async updateNFTHistory() {
       this.isHistoryInfoLoading = true;
-      let hasHistoryInDatabase = false;
+      const historyOnChain = await this.getNFTEventsAll();
+      this.NFTHistory = historyOnChain;
+
       if (this.isWritingNFT) {
```

Comment:
The history UI might flash twice if `isWritingNFT` is `true`, since `this.NFTHistory` is assigned here and assigned again later after `isWritingNFT` check

Should assign `this.NFTHistory` after `isWritingNFT` check